src/client/Client.cc

```diff
@@ -4425,7 +4425,7 @@ int Client::_do_remount(bool retry_on_error)
       !(retry_on_error && (++retries_on_invalidate < max_retries));
     if (should_abort && !is_unmounting()) {
       lderr(cct) << "failed to remount for kernel dentry trimming; quitting!" << dendl;
-      ceph_abort();
+      should_abort_on_failure = true;
```

Comment:
As a best practice, if the passed in argument can be modified, it should be passes as a pointer.

Also, what do you think about returning a std::pair<> (rather than passing in a bool) from this function?

Reply:
To return std::pair<int, bool> rather than adding parameters to function make sense.